specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/virtualNetworkGateway.json

```diff
@@ -2385,6 +2619,87 @@
       ],
       "description": "IP configuration for virtual network gateway."
     },
+    "VirtualNetworkGatewayNatRuleProperties": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the NAT Rule resource."
+        },
+        "type": {
```

Comment:
top level property names must not be repeated under "properties" envelope

Reply:
rename it natType or something like that.

Reply:
Actually It is already defined with this name in NRP. Should we change property name here. it will be a mismatch between actual object and swagger-         /// <summary>
        /// Gets or sets the NAT type
        /// </summary>
        public VpnNatRuleType Type { get; set; }

Reply:
For future reference, Please note that such changes are not allowed. APIs must be reviewed before service code is deployed.

Reply:
Sure Thank you so much I will take this as a learning and keep in mind for future.